codebuild/codebuild_validate_terraform/code_build_validate_terraform.yml

```diff
@@ -0,0 +1,30 @@
+version: 0.2
+
+env:
+  shell: bash
+
+phases:
+  pre_build:
+    commands:
+      - echo '[INFO] Validate Terraform...'
+      - cd $TERRAFORM_DIRECTORY
+      - terraform init --backend=false
+      - terraform validate
+  build:
+    commands:
+      - echo '[INFO] Assuming TF Deployer IAM role...'
+      - role_arn="arn:aws:iam::$AWS_ACCOUNT_ID:role/$ROLE_NAME"
+      - source /usr/local/bin/sts-assume-role.sh "$role_arn" "$AWS_DEFAULT_REGION" $STS_ASSUME_ROLE_DURATION
+      - if [ -f "$BACKEND_VAR_FILE" ]; then
+          terraform init -backend="$BACKEND_VAR_FILE";
+        else
+          terraform init;
+        fi
+      - terraform plan --out=/opt/terraform_$AWS_ACCOUNT_ID.plan
+
+artifacts:
+  name: TerraformPlanArtifacts
+  files:
+    - terraform_$AWS_ACCOUNT_ID.plan
```

Comment:
Have you checked that an env var here gets populated and should there be an optional name var that gets passed into this name as well so if there are multiple terraforms in the same account it names each one differently?

Reply:
This is what I wasn't sure about, should a map of AWS account IDs be passed in as a variable when calling the module, and then iterate over the set using `for_each = var.aws_accounts` ?

e.g 

```
aws_accounts = {
    staging_account = {
      environment    = "test"
      aws_account_id = 103495720024
    }

    # prod_account = {
    #   environment    = "prod"
    #   aws_account_id = 779799343306
    # }
  }
```

environment variable in codebuild project which runs Terraform init & validate -:

```
environment_variable {
      name  = "AWS_ACCOUNT_ID"
      value = each.value.aws_account_id
    }
```

Reply:
I wasn't really thinking about multiple accounts but good point. I was thinking about different terraform deployments. So the first one I'm thinking of is the cloudwatch config where there are 2 separate terraforms. One that gets deployed to test and prod and 1 that gets deployed into all of our accounts. 

I was just thinking if the filename should be `terraform_plan_[account]_[purpose]` where if purpose isn't specified it defaults to `main` or `default` or something.